sourcegraph/sourcegraph.proto

```diff
@@ -288,58 +289,115 @@ message RepoList {
 
 service RepoBadges {
 	// ListBadges lists the available badges for repo.
-	rpc ListBadges(RepoSpec) returns (BadgeList);
+	rpc ListBadges(RepoSpec) returns (BadgeList) {
+		option (google.api.http) = {
+			get: "/v1/repo_badges/list_badges"
```

Comment:
> What use does the version keep if it's optional? Do we plan to keep full backwards compatibility for the API forever?

API version parameter being optional is an orthogonal decision. I just meant if you don't specify, it automatically picks the latest version. If you care about locking down to a specific version, you have to specify.

We can make it mandatory and just return an error if you don't specify any version. But I thought it'd be nice to just opt to use latest version unless specified (similar in spirit to how `go get -u` works).

> My rational behind this was that Twitter, Tumblr, and various other services all do it this way. I think the intent of placing it inside the URL is that removal of the v1 API completely would have users hitting 404 not found, rather than accidentally running into a new API URL under the same name.

Hmm, okay. I am okay with it either way.

> rather than accidentally running into a new API URL under the same name.

I think we'd want to avoid that situation in either case. I think the same applies for protobufs. If you want to change something, you have to create a new field (with a different name) and delete/deprecate the old one, rather than reusing it.